tempodb/tempodb.go

```diff
@@ -544,3 +544,33 @@ func (rw *readerWriter) compactedBlocklist(tenantID string) []*encoding.Compacte
 
 	return copiedBlocklist
 }
+
+// updateBlocklist Add and remove regular or compacted blocks from the in-memory blocklist.
+// Changes are temporary and will be overwritten on the next poll.
+func (rw *readerWriter) updateBlocklist(tenantID string, add []*encoding.BlockMeta, remove []*encoding.BlockMeta, compactedAdd []*encoding.CompactedBlockMeta) {
+	if tenantID == "" {
+		return
+	}
+
+	rw.blockListsMtx.Lock()
+	defer rw.blockListsMtx.Unlock()
+
+	// ******** Regular blocks ********
+	toRemove := make(map[*encoding.BlockMeta]bool)
+	for _, m := range remove {
+		toRemove[m] = true
+	}
+
+	blocklist := rw.blockLists[tenantID]
+	newblocklist := make([]*encoding.BlockMeta, 0, len(blocklist)-len(remove)+len(add))
+	for _, m := range blocklist {
+		if _, ok := toRemove[m]; !ok {
```

Comment:
should we compare by pointer?  would by block guid make more sense/be possible?

Reply:
Agree, changed to compare block meta by ID.